src/v/cluster_link/replication/deps.h

```diff
@@ -56,6 +56,14 @@ class data_sink {
       = 0;
 
     virtual kafka::offset start_offset() = 0;
+
+    // If needed, push the highest seen producer ID into id_allocator. This
+    // helps avoid false-positive idempotency errors, e.g. if, after failing
+    // over, a new producer on the target cluster were assigned an ID already
+    // used on the source cluster. We can avoid this by always making sure the
+    // next ID persisted to the allocator is ahead of the highest producer ID
+    // we've seen from a source cluster batch.
+    virtual ss::future<> maybe_sync_pid() = 0;
```

Comment:
can this be an implementation detail? (as in can this be a part of local_partition_sink impl directly and avoid updating the interface here), seems too specific to be here.

Reply:
talked offline - TL;DR replicate is not an async function, so in order to do this we have to change the API, regardless. Seems less invasive to just bolt on this extra function (vs making ::replicate itself async)